app/vmselect/prometheus/prometheus.go

```diff
@@ -677,7 +685,12 @@ func SeriesHandler(qt *querytracer.Tracer, startTime time.Time, at *auth.Token,
 	if err != nil {
 		return err
 	}
-	sq := storage.NewSearchQuery(at.AccountID, at.ProjectID, cp.start, cp.end, cp.filterss, *maxSeriesLimit)
+
+	minLimit := *maxSeriesLimit
+	if limit > 0 && limit < *maxSeriesLimit {
+		minLimit = limit
+	}
+	sq := storage.NewSearchQuery(at.AccountID, at.ProjectID, cp.start, cp.end, cp.filterss, minLimit)
```

Comment:
I'm unsure whether this new logic is correct here :( It needs additional investigation.

Reply:
can try？

Reply:
> I'm unsure whether this new logic is correct here :( It needs additional investigation.

I understand ok，can reduce unnecessary vmstorage query